src/lib/utils/utils.js

```diff
@@ -90,44 +90,57 @@ function getJson(url, cb) {
 }
 
 function pingEndpoint(host, port, type, protocol, origin, callback) {
-  const options = {
-    protocolVersion: 13,
-    perMessageDeflate: true,
-    origin: origin,
-    host: host,
-    port: port
+  // remove trailing api key from infura, e.g. rinkeby.infura.io/nmY8WtT4QfEwz2S7wTbl
+  const _host = host.indexOf('/') > -1 ? host.split('/')[0] : host;
```

Comment:
This comment and code were carried over from the previous implementation, but it does need another look since `host` could contain a a `:[port]` section and when the websocket connection is constructed it would be an invalid url. Thanks for raising my attention!

Reply:
See: https://github.com/embark-framework/embark/pull/1196/commits/5a169031a76e7ebba4867639bf336f4321149daa

I made it more generic. Note that if a ping really should be tried with a path other than `/` (maybe with query parameters), then `pingEndpoint` would need to allow for an additional `path` parameter, or maybe an `options` parameter-object that could include `path:`.